wire-ios-sync-engine/Source/SessionManager/SessionManager+Backup.swift

```diff
@@ -56,9 +55,17 @@ extension SessionManager {
             applicationContainer: sharedContainerURL,
             dispatchGroup: dispatchGroup,
             databaseKey: activeUserSession.managedObjectContext.databaseKey,
-            completion: { [dispatchGroup] in
+            completion: { [dispatchGroup] result in
+                switch result {
+                case .success:
+                    break
+                case .failure:
+                    let exportBackupFailed = BackupExportFailedAnalyticsEvent()
+                    activeUserSession.analyticsSession?.trackEvent(exportBackupFailed)
```

Comment:
so we'll always need two steps for sending events:
1. create the event
2. pass it to trackEvent

even if the trackedEvent does not require any argument.

I would have expect something more straightforward like :
```
activeUserSession.analyticsSession?.trackEvent(.backupExportFailed)
```

but maybe I am missing the advantage of structs over enums

Reply:
In this case, using structs over enums have certain advantages in our case:

1.	**Extensibility**: Structs allow easy addition of properties and methods, enhancing flexibility.
2.	**Type Safety and Clarity**: Encapsulate all event-related data in a single, clear structure.
3.	**Protocol Conformance**: Enables uniform handling of different event types by conforming to protocols. In which we do conform. All events are conforming to 

```swift
public protocol AnalyticsEvent {

    var eventName: String { get }
    var segmentation: [String: String] { get }

}
```

4.	**Mutability**: Offers the choice of immutability (let) or mutability (var).
5.	**Encapsulation**: Combines data and behavior in one cohesive unit.
6.	**Single Responsibility**: Separates event data definition from event outcome definition, adhering to the single responsibility principle.

Reply:
Perhaps we can have both:

```swift
extension AnalyticsEvent {

  static let backupSuccess: some AnalyticsEvent {
    BackupSuccessAnalyticsEvent()
  }

  static func someEventWithSegmentation(foo: Int) -> some AnalyticsEvent {
    SomeEvent(foo: foo)
  }

}

analyticsSession.trackEvent(.backupSuccess)
analyticsSession.trackEvent(.someEventWithSegmentation(foo: 42))
```

Reply:
Or just write 
```suggestion
                    activeUserSession.analyticsSession?.trackEvent(BackupExportFailedAnalyticsEvent())
```